src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -135,9 +139,53 @@ public Optional<Address> getRetiringFederationAddress() {
         return federatorSupport.getRetiringFederationAddress();
     }
 
+    @Override
+    public Optional<Federation> getProposedFederation() {
+        if (!federatorSupport.getConfigForBestBlock().isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+       
+        Integer federationSize = federatorSupport.getProposedFederationSize()
+            .orElse(FEDERATION_NON_EXISTENT.getCode());
+        if (federationSize == FEDERATION_NON_EXISTENT.getCode()) {
+            return Optional.empty();
+        }
+
+        List<FederationMember> federationMembers = IntStream.range(0, federationSize)
+            .mapToObj(i -> new FederationMember(
+                federatorSupport.getProposedFederatorPublicKeyOfType(i, KeyType.BTC)
+                    .map(ECKey::getPubKey)
+                    .map(BtcECKey::fromPublicOnly)
+                    .orElseThrow(IllegalStateException::new),
+                federatorSupport.getProposedFederatorPublicKeyOfType(i, KeyType.RSK)
+                    .orElseThrow(IllegalStateException::new),
+                federatorSupport.getProposedFederatorPublicKeyOfType(i, KeyType.MST)
+                    .orElseThrow(IllegalStateException::new)
+            ))
+            .toList();
+
+        FederationArgs federationArgs = new FederationArgs(
+            federationMembers,
+            federatorSupport.getProposedFederationCreationTime()
+                .orElseThrow(IllegalStateException::new),
```

Comment:
Where is this exception being caught (if so)? I think we should add a context message when instantiating it or when catching it. Wdyt?

Reply:
True, I have wrapped in a try-catch with logging the error.